src/zinolib/utils.py

```diff
@@ -56,3 +58,27 @@ def generate_authtoken(challenge, password):
     raw_token = "%s %s" % (challenge, password)
     token = hashlib.sha1(raw_token.encode("UTF-8")).hexdigest()
     return token
+
+
+def log_exception_with_params(logger, reraise=True, return_value=None):
+    """Log the params and exception if the wrapped function fails
+
+    If ``reraise`` is False, return ``return_value`` instead of reraising the
+    exception.
+    """
+    def inner(function):
+        @functools.wraps(function)
+        def wrapper(*args, **kwargs):
+            params = f'args={args} kwargs={kwargs}'
```

Comment:
Why spend the CPU cycles building this string representation before we even run the wrapped function, if we expect exceptions to be quite rare in production?